osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -171,7 +176,22 @@ public void SelectRandom()
             if (visibleGroups.Count < 1)
                 return;
 
-            BeatmapGroup group = visibleGroups[RNG.Next(visibleGroups.Count)];
+            BeatmapGroup group;
+            if (randomType == SelectionRandomType.RandomPermutation)
+            {
+                List<BeatmapGroup> notSeenGroups = visibleGroups.Except(seenGroups).ToList();

```

Comment:
Actually, nevermind. Revisiting the surrounding code I think I prefer your solution quite a bit. :)